examples/docs_snippets/docs_snippets/guides/components/shell-script-component/with-build-defs.py

```diff
@@ -7,28 +7,27 @@
     AssetSpecModel,
     Component,
     ComponentLoadContext,
-    ResolvableModel,
-    ResolvedFrom,
+    Resolvable,
 )
 from dagster_components.resolved.core_models import ResolvedAssetSpec
 
 import dagster as dg
 
 
-class ShellCommandModel(ResolvableModel):
-    script_path: str
-    asset_specs: Sequence[AssetSpecModel]
-
-
 @dataclass
-class ShellCommand(Component, ResolvedFrom[ShellCommandModel]):
+class ShellCommand(Component, Resolvable):
     """Models a shell script as a Dagster asset."""
 
-    script_path: str
-    asset_specs: Sequence[ResolvedAssetSpec]
+    def __init__(
+        self,
+        script_path: str,
+        asset_specs: Sequence[ResolvedAssetSpec],
+    ):
+        self.script_path = script_path
+        self.asset_specs = asset_specs
```

Comment:
is this necessary with `@dataclass`?

Reply:
no we could just do `@dataclass`
think we will just follow what we decide in https://github.com/dagster-io/dagster/pull/28821